R/src_gen.R

```diff
@@ -0,0 +1,207 @@
+    hand[is.na(hand[])] <- 0
+    # Generate Slope Raster
+    slope <- raster::terrain(hand)
+    slope[is.na(slope[])] <- 0
+    # Get raster cell resolution
+    res <- compute_cellres(hand)
+    # Create HAND_Slope data frame
+
+    if (progress) pb$tick()
+
+    hand_slope <-
+        data.frame(
+            hand = raster::getValues(hand),
+            slope = raster::getValues(slope)
+        ) %>%
+        na.omit() %>%
+        dplyr::arrange(hand) %>%
+        dplyr::mutate(
+            res = res,
+            BA  = res * res * sqrt(1 + slope)
+        )
+
+    if (progress) pb$tick()
+
+    # Get VAAs via nhdplusTools
+    vaa <-
+        nhdplusTools::get_vaa(
+            atts = c(
+                "pathlength",
+                "arbolatesu",
+                "lengthkm",
+                "areasqkm",
+                "slope",
+                "roughness"
+            )
+        ) %>%
+        dplyr::filter(comid %in% comids)
+
+    if (progress) pb$tick()
+
+    synthetic_rating_curve <-
+        lapply(
+            stage,
+            FUN = function(y) {
+
+                if (progress) pb$tick()
+
+                vaa %>%
+                    na.omit() %>%
+                    dplyr::group_by(comid) %>%
+                    dplyr::mutate(
+                        n_prod = sqrt(slope) / ((lengthkm * 1000) * roughness),
+                        Y = y
+                    ) %>%
+                    dplyr::ungroup() %>%
+                    dplyr::filter(dplyr::across(.fns = is.finite)) %>%
+                    data.table::as.data.table() %>%
+                    .[,
+                      Q := compute_flat_tub(y, hand_slope) * n_prod,
+                      by = .(comid)
+                     ]
+            }
+        ) %>%
+        data.table::rbindlist() %>%
+        tibble::as_tibble() %>%
+        dplyr::select(comid, Y, Q) %>%
+        dplyr::rename(COMID = comid)
+
+    synthetic_rating_curve
+}
+
+#' @title Get catchmask raster via nhdplusTools
+#' @param aoi An area of interest. Can be retrieved via \code{AOI::aoi_get}.
+#' @param template A \code{raster} to template
+#'                 the catchmask raster to.
+#' @return A \code{raster} representing the Catchmask
+#'         templated to **template**.
+#' @importFrom fasterize fasterize
+#' @importFrom nhdplusTools get_nhdplus
+#' @keywords internal
+#' @export
+get_catchmask <- function(aoi, template) {
+    this_nhd <- nhdplusTools::get_nhdplus(
+                    aoi,
+                    realization = "catchment"
+                ) %>%
+                sf::st_as_sf() %>%
+                sf::st_cast("MULTIPOLYGON")
+
+    fasterize::fasterize(
+        this_nhd,
+        template,
+        field = "featureid"
+    )
+}
+
+#' @title Get roughness value prediction
+#'
+#' @description
+#' This function querys a REST API serving the Gradient Boosting machine
+#' learning model developed in:
+#'
+#' **Johnson, J.M., Eyelade D., Clarke K.C, Singh-Mohudpur, J. (2021)
```

Comment:
need to say in review, not yet published

Reply:
Done. Will show on next commit 👍